rust/lance/src/dataset/fragment.rs

```diff
@@ -207,19 +208,50 @@ impl GenericFileReader for FileReader {
     }
 }
 
+#[derive(Debug, Clone)]
+struct V2Reader {
```

Comment:
1. can this move to `lance-io`?
2. lets just use a `mod v2 { struct Reader }` so that `V2` will not leave in the codebase forever?

Reply:
1.  No, `lance-io` is too low level.  It doesn't know about the file readers.  I could maybe move it to `lance-file` but at the moment I'd rather keep it here.  This type exists to adapt the v2 reader to `GenericFileReader` and `GenericFileReader` is currently "the methods that a lance dataset expects from a file reader" so it makes sense to me that the trait / impl / adapters exist at the dataset level.
2. Done.